src/mds/StrayManager.h

```diff
@@ -121,77 +169,29 @@ class StrayManager
 
   void _eval_stray_remote(CDentry *stray_dn, CDentry *remote_dn);
 
-  // My public interface is for consumption by MDCache
-  public:
-  explicit StrayManager(MDSRank *mds, PurgeQueue &purge_queue_);
-  void set_logger(PerfCounters *l) {logger = l;}
-  void activate();
-
-  bool eval_stray(CDentry *dn);
-
-  void set_num_strays(uint64_t num);
-  uint64_t get_num_strays() const { return num_strays; }
+  // Has passed through eval_stray and still has refs
+  elist<CDentry*> delayed_eval_stray;
 
-  /**
-   * Queue dentry for later evaluation. (evaluate it while not in the
-   * middle of another metadata operation)
-   */
-  void queue_delayed(CDentry *dn);
+  // strays that have been trimmed from cache
+  std::set<std::string> trimmed_strays;
 
-  /**
-   * Eval strays in the delayed_eval_stray list
-   */
-  void advance_delayed();
+  // Global references for doing I/O
+  MDSRank *mds;
+  PerfCounters *logger;
 
-  /**
-   * Remote dentry potentially points to a stray. When it is touched,
-   * call in here to evaluate it for migration (move a stray residing
-   * on another MDS to this MDS) or reintegration (move a stray dentry's
-   * inode into a non-stray hardlink dentry and clean up the stray).
-   *
-   * @param stray_dn a stray dentry whose inode has been referenced
-   *                 by a remote dentry
-   * @param remote_dn (optional) which remote dentry was touched
-   *                  in an operation that led us here: this is used
-   *                  as a hint for which remote to reintegrate into
-   *                  if there are multiple remotes.
-   */
-  void eval_remote(CDentry *remote_dn);
+  bool started;
 
-  /**
-   * Given a dentry within one of my stray directories,
-   * send it off to a stray directory in another MDS.
-   *
-   * This is for use:
-   *  * Case A: when shutting down a rank, we migrate strays
-   *    away from ourselves rather than waiting for purge
-   *  * Case B: when a client request has a trace that refers to
-   *    a stray inode on another MDS, we migrate that inode from
-   *    there to here, in order that we can later re-integrate it
-   *    here.
-   *
-   * In case B, the receiver should be calling into eval_stray
-   * on completion of mv (i.e. inode put), resulting in a subsequent
-   * reintegration.
-   */
-  void migrate_stray(CDentry *dn, mds_rank_t dest);
+  // Stray dentries for this rank (including those not in cache)
+  uint64_t num_strays;
```

Comment:
You can move the initialization from the `.cc` file to this header.

Reply:
I have moved them in a new commit.